test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -1,181 +1,84 @@
-
-    def test_remove_custom_removes_custom_added_value(self):
-        self.merge_dict.set_base("a", 42)
-        self.merge_dict.set_base("b", 0)
-        self.merge_dict.add_custom("a", 100)
-        self.merge_dict.add_custom("c", 1)
-
-        self.assertEqual(self.merge_dict["a"], {42, 100})
-        self.assertEqual(self.merge_dict["c"], {1})
-
-        self.merge_dict.remove_custom("a")
-        self.merge_dict.remove_custom("c")
-        self.assertEqual(self.merge_dict["a"], {42})
-        self.assertEqual(self.merge_dict.get("c"), None)
-        self.assertFalse(self.merge_dict.custom_added("a"))
-        self.assertFalse(self.merge_dict.custom_added("c"))
-
-    def test_remove_custom_removes_custom_added_key(self):
-        self.merge_dict.add_custom("a", 100)
-        self.assertEqual(self.merge_dict["a"], {100})
-        self.assertEqual(len(self.merge_dict), 1)
-        self.merge_dict.add_custom("a", 1001)
-        self.assertEqual(self.merge_dict["a"], {100, 1001})
-        self.assertEqual(len(self.merge_dict), 1)
-        # NOTE: remove_custom removes the whole overloads
-        self.merge_dict.remove_custom("a")
-        self.assertEqual(len(self.merge_dict), 0)
-        self.assertNotIn("a", self.merge_dict)
-
-    def test_bool_is_true_when_not_empty(self):
-        if self.merge_dict:
-            self.fail("MergeDict should be false when empty")
-        self.merge_dict.add_custom("a", 1)
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-        self.merge_dict.set_base("a", 42)
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-        self.merge_dict.remove_custom("a")
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-
-
-class TestRegistrationDecorators(common_utils.TestCase):
+
+class TestRegistration(common_utils.TestCase):
     def setUp(self) -> None:
         self.registry = registration.OnnxRegistry()
+        self.custom_domain = onnxscript.values.Opset(domain="custom", version=1)
 
     def tearDown(self) -> None:
         self.registry._registry.pop("test::test_op", None)
 
-    def test_onnx_symbolic_registers_function(self):
-        self.assertFalse(self.registry.is_registered_op("test::test_op", 9))
+    def test_onnx_custom_symbolic_registers_function(self):
+        self.assertFalse(self.registry.is_registered_op("test", "test_op", "default"))
 
-        def test(g, x):
-            return g.op("test", x)
-
-        self.registry.register("test::test_op", 9, test, custom=True)
-        self.assertTrue(self.registry.is_registered_op("test::test_op", 9))
-        function_group = self.registry.get_function_group("test::test_op")
-        assert function_group is not None
-        self.assertEqual(function_group.get(9), {test})
+        @onnxscript.script(self.custom_domain)
+        def test(x, y):
```

Comment:
```suggestion
        def custom_add(x, y):
```

I would avoid overly generic names